src/builtins/BuiltinTemporal.cpp

```diff
@@ -29,6 +29,14 @@
 namespace Escargot {
 #if defined(ESCARGOT_ENABLE_TEMPORAL)
 
+#define CHECK_TEMPORAL_OBJECT_HAS_YEAR_AND_MONTH(temporalDateLike) \
+    !(temporalDateLike.isObject() && (temporalDateLike.asObject()->isTemporalPlainDateObject() || temporalDateLike.asObject()->isTemporalPlainDateTimeObject() || temporalDateLike.asObject()->isTemporalPlainYearMonthObject()))
```

Comment:
Is it worth to add some virtual function which checks this in one step? Especially if it is used frequently.
The helper could return an enum, if similar group checks exists.

Reply:
Yes i will do a virtual function for year, month and calendar. Currently these 3 are needed.